controllers/managedfusionoffering_controller.go

```diff
@@ -84,11 +122,20 @@ func (r *ManagedFusionOfferingReconciler) Reconcile(ctx context.Context, req ctr
 		return result, fmt.Errorf("an error was encountered during reconcilePhases: %v", err)
 	}
 
-	if result, err := pluginReconcile(r, r.managedFusionOffering); err != nil {
-		return ctrl.Result{}, fmt.Errorf("An error was encountered during reconcile: %v", err)
+	if !r.DoesCRDExistsAtStart {
+		if doesCRDsExists() {
+			r.Log.Info("Restarting to add watches for offering resources")
+			os.Exit(1)
+		}
 	} else {
-		return result, nil
+		if result, err := pluginReconcile(r, r.managedFusionOffering); err != nil {
+			return ctrl.Result{}, fmt.Errorf("An error was encountered during reconcile: %v", err)
+		} else {
+			return result, nil
+		}
 	}
+
+	return ctrl.Result{}, nil
```

Comment:
All of this code is very specific for DF, even if it does not say DF in the variable names, we should not do that. For this workaround, only the actual offering should know if it needs a restart or not and that code should go into the dfReconcile code at the start. But the reconciler should implement an `r.Exist()` function to allow the offering to exit without using a hack.  
```suggestion
	if result, err := pluginReconcile(r, r.managedFusionOffering); err != nil {
		return ctrl.Result{}, fmt.Errorf("An error was encountered during reconcile: %v", err)
	} else {
		return result, nil
	}
```